app/controllers/reservations_controller.rb

```diff
@@ -394,6 +394,20 @@ def upcoming_today?(reservation)
     (reservation.reserve_start_at.to_date == now.to_date || reservation.reserve_start_at < now) && reservation.reserve_end_at > now
   end
 
+  # Some browsers are not updating their cached JS and have an out-of-date calendar
+  # JS library which uses unix timestamps for the parameters. This allows us to handle both.
+  def parse_time_param(value)
+    return unless value
+
+    if value.to_s.match?(/\A\d{10}\z/) # unix timestamp
+      Time.zone.at(value.to_i)
+    elsif value.to_s.match?(/\A\d{13}\z/) # unix timestamp with milliseconds
+      Time.zone.at(value.to_i / 1000)
+    else
+      Time.zone.parse(value)
+    end
+  end
```

Comment:
I think that instead of checking regular expressions here and trying to guess how to process the timestamp, it may be better to do something like this:

```
def parse_time_param(string_value)
  return unless string_value
  Time.zone.parse(string_value)
rescue ArgumentError
  Time.at(string_value.to_i)
end
```

That way, we’re only falling back to `Time.at` if `Time.parse` fails, accomodating UNIX timestamps as well, without making assumptions about the format of the string.

Reply:
Yeah, I thought about that, and was all "don't use exceptions for flow control", but in this case I think it probably makes more sense.